### PR TITLE
Port Anime4k(Upscale_Denoise_CNN_x2_S) and utilizes multi-pass shaders serving as a JS example

### PR DESCRIPTION
**Pull request by adglkh**
- feat(js-sdk): cache uniforms and attributes during shader loading
- feat(js-sdk): provide an additional uniform for the source video texture
- feat(js-sdk): add a js example based on Anime4K upscaling and denoising shaders

### CODE REVIEW

File: examples/js/sgsr.html [hunk 1]
```diff
@@ -0,0 +1,218 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta content="text/html;charset=utf-8" http-equiv="Content-Type">
+    <meta content="utf-8" http-equiv="encoding">
+    <title>Anbox Streaming SDK Example - Enable video upscaling with Snapdragon™ Game Super Resolution(SGSR) </title>
```
Comment (ajanon) [inline examples/js/sgsr.html:6]: Random question: do we need an agreement from them to use this name?
Or put a disclaimer (e.g. 'This trademark is possessed by...')

**Review by ajanon — APPROVED** [review]
LGTM, one small question

File: examples/js/sgsr.html (hunk 1 above)
Comment (ajanon) [reply examples/js/sgsr.html:6]: This much better fits the other PR actually, let me put a comment there.

Comment (adglkh) [reply examples/js/sgsr.html:6]: Thanks @ajanon, would you please take another look?

**Review by morphis — APPROVED** [review]
LGTM

**Pull request closed without merging**